File: JavaAdvancedExams/JavaFundamentalsExam23April2016/src/MelrahShake.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;

public class MelrahShake {
    public static void main(String[] args) throws IOException {
        BufferedReader reader = new BufferedReader(new InputStreamReader(System.in));

        StringBuilder input = new StringBuilder(reader.readLine());
        StringBuilder pattern = new StringBuilder(reader.readLine());

        while (input.indexOf(pattern.toString()) != -1
                && input.indexOf(pattern.toString()) != input.lastIndexOf(pattern.toString())
                && pattern.length() != 0) {
            int firstIndex = input.indexOf(pattern.toString());
            input.delete(firstIndex, firstIndex + pattern.length());

            int lastIndex = input.lastIndexOf(pattern.toString());
            input.delete(lastIndex, lastIndex + pattern.length());

            pattern.deleteCharAt(pattern.length() / 2);

            System.out.println("Shaked it.");
        }

        System.out.println("No shake.");

        System.out.println(input);
    }
}

File: TechModule/ProgrammingFundamentals/ProgrammingFundamentalsExercises/StringsAndRegExExercises/09.MelrahShake/MelrahShake.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

class MelrahShake
{
    static void Main(string[] args)
    {
        string input = Console.ReadLine();
        string pattern = Console.ReadLine();

        while (input.IndexOf(pattern) != input.LastIndexOf(pattern) && pattern.Length > 0)
        {
            Console.WriteLine("Shaked it.");
            input = input.Remove(input.IndexOf(pattern), pattern.Length);
            input = input.Remove(input.LastIndexOf(pattern), pattern.Length);
            pattern = pattern.Remove(pattern.Length / 2, 1);
        }

        Console.WriteLine("No shake.");
        Console.WriteLine(input);
    }
}